sysvabi64/sysvabi64.rst

```diff
@@ -1592,6 +1592,163 @@ IFUNC requirements for dynamic linkers
 To resolve an ``R_AARCH64_IRELATIVE`` relocation the dynamic linker
 performs the calculation described in AAELF64_ Dynamic Relocations.
 
+Function Multi-versioning
+-------------------------
+
+Function Multi-versioning (FMV) is an Arm C Language Extension that
+lets the compiler generate multiple function versions and auto-dispatch
+between them. Each of the function versions is specialized for a set
+of architecture extensions. The most suitable version is selected
+dynamically. This requires runtime information about the CPU features
+available on the host. FMV is supported on GNU/Linux, Android, and
+many of the BSD operating systems.
+
+On System V platforms Function Multi-versioning is implemented using
+GNU Indirect Functions (IFUNC). The compiler generated IFUNC resolver
+may rely on a runtime library for selecting the most suitable function
+version, but it is not a requirement.
+
+If a runtime library is used, it must ensure that a global variable
```

Comment:
I think we should take this part out if all SystemV platforms use IFUNCS. It is already covered by the paragraph above. For ifuncs to work ``__aarch64_cpu_features`` needs to be populated before it is used by the IFUNC resolver and that is before main and before static constructors are run.